src/basic/architecture.c

```diff
@@ -176,3 +178,228 @@ static const char *const architecture_table[_ARCHITECTURE_MAX] = {
 };
 
 DEFINE_STRING_TABLE_LOOKUP(architecture, Architecture);
+
+Abi preferred_abi(void) {
+
+        /* Return a usable enum identifying the architecture we are running on. This
+         * is based on uname(), and the user may hence control what this returns by using
+         * personality(). This puts the user in control on systems that can run binaries
+         * of multiple architectures.
+         *
+         * We do not translate the string returned by uname() 1:1. Instead we try to
+         * clean it up and break down the confusion on x86 and arm in particular.
+         *
+         * We try to distinguish CPUs, not CPU features, i.e. actual architectures that
+         * have genuinely different code. This includes different userspace ABI, as used
+         * for LIB_ARCH_TUPLE and defined in https://wiki.debian.org/Multiarch/Tuples as
+         * ooposed to uname_architecture() above, which is solely and exclusively for CPU
+         * comparisons. In most cases, especially when comparing images, you want to use
+         * this instead. */
```

Comment:
"this instead" will not be clear without context. Maybe say "this instead of Architecture".